### DISCUSSION TITLE
How do I access the metadata from LUA script?

### DISCUSSION THREAD

**Discussion by fmhwong** [fluent/fluent-bit#9431] [Q&A] [answered]
In the LUA script, the method signature is always tag, timestamp, record.

```
function test(tag, timestamp, record)
    ...
    return 1, timestamp, record
end
```

I can access the log body from the `record` but I can't access to the metadata.  How can I get the metadata in the above function?  I'd like to read the metadata and append it to the body.

**Comment by patrick-stephens** [comment]
It is not possible currently but is being worked on for a future release.

**Reply by fmhwong** [reply]
Do you have an issue open for this work?

**Comment by zane-deg** [comment, marked as answer]
For anyone else arriving here from search this is now possible: https://docs.fluentbit.io/manual/data-pipeline/filters/lua#extended-function-signature

**Reply by patrick-stephens** [reply]
Yeah I lost track!
